src/site/content/en/blog/module-workers/index.md

```diff
@@ -0,0 +1,229 @@
+loading also caches parsed code, which means modules that are used on the main thread and in a
+Worker only need to be parsed once.
+
+Moving to JS Modules also enables the use of [dynamic
+import](https://v8.dev/features/dynamic-import) for lazy-loading code without blocking execution of
+the worker. Dynamic Import is much more explicit than using importScripts to load dependencies,
+since the imported module's exports are returned rather than relying on global variables.
+
+##### worker.js:
+
+```js
+import { sayHello } from './greet.js';
+addEventListener('message', e => {
+  postMessage(sayHello());
+});
+```
+
+##### greet.js:
+
+```js
+import greetings from './data.js';
+export function sayHello() {
+  return greetings.hello;
+}
+```
+
+To ensure great performance, the old `importScripts()` API is not available within Module Workers.
+Switching Workers to use JS Modules means all code is loaded in [strict
+mode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode), and
+HTML-style comments are not supported. Another notable change is that the value of `this` in the
+top-level scope of a JS Module is `undefined`, whereas in Classic Workers the value was the
+worker's global scope. Fortunately, there has always been a `self` global that provides a reference
+to the global scope - it's available in all types of Workers including Service Workers, as well as
+in the DOM.
+
+## Preload Workers with modulepreload
+
+One substantial performance improvement that comes with Module Workers is the ability to preload
+workers and their dependencies. With Module Workers, scripts are loaded and executed as standard
+JavaScript Modules, which means they can be preloaded and even pre-parsed using `modulepreload`:
+
+```html
+<!-- preloads worker.js and its dependencies: -->
+<link rel="modulepreload" href="worker.js">
+
+<script>
+  addEventListener('load', () => {
+    // our worker code is likely already parsed and ready to execute!
+    const worker = new Worker('worker.js', { type: 'module' });
+  });
+</script>
+```
+
+Preloaded modules can also be used by both the main thread and Module Workers. This is useful for
+modules that are imported in both contexts, or in cases where it's not possible to know in advance
+whether a module will be used on the main thread or in a Worker.
+
+Previously, the options available for preloading Web Worker scripts were limited and not
+necessarily reliable. Classic Workers had their own "worker" resource type for preloading, but no
+browsers implemented `<link rel="preload" as="worker">`. As a result, the primary technique
+available for preloading Worker scripts was to use `<link rel="prefetch">`, which relied entirely
+on the HTTP cache. When used in combination with caching headers were set up correctly, this made
+it possible to avoid `new Worker()` having to wait to download the Worker script, but did not
```

Comment:
Change '`new Worker()`' to 'worker instantiation'. It's a little more formal, which is more appropriate for this content channel.